cli/internal/context/context.go

```diff
@@ -536,11 +536,12 @@ func calculateGlobalHash(rootpath string, rootPackageJSON *fs.PackageJSON, exter
 
 	if !util.IsYarn(backend.Name) {
 		// If we are not in Yarn, add the specfile and lockfile to global deps
-		globalDeps.Add(backend.Specfile)
-		globalDeps.Add(backend.Lockfile)
+		globalDeps.Add(filepath.Join(rootpath, backend.Specfile))
+		globalDeps.Add(filepath.Join(rootpath, backend.Lockfile))
```

Comment:
I think the absolute path will break caching across machines? Is there a different approach we can use?

Reply:
Good point. I suppose we can do everything relative to the repo root instead?

Reply:
I think so. We might need to normalize paths on windows though?

Reply:
Yup. We probably have this problem with `git ls-tree` too.

Reply:
I made #891 a separate PR to normalize paths coming from `git ls-tree`